ja/lessons/advanced/nerves.md

```diff
@@ -0,0 +1,262 @@
+```
+
+これがあなたが書き込むカードであると確信しているなら、 `Y`を選んで、しばらくするとメモリカードは準備ができています:
+
+```
+Use 7.42 GiB memory card found at /dev/rdisk2? [Yn]
+|====================================| 100% (32.51 / 32.51) MB
+Success!
+Elapsed time: 8.022 s
+```
+
+これで、メモリカードをデバイスに挿入して動作するかどうかを確認します。
+
+画面が接続されている場合、このメモリカードを挿入した状態でデバイスの電源を入れた後に、Linuxの起動シーケンスが表示されるはずです。
+
+## ネットワークの設定
+
+次のステップはネットワークを設定することです。Nervesのエコシステムはさまざまなパッケージを提供しています。[nerves_network](https://github.com/nerves-project/nerves_network) は、有線イーサネットポートを介してデバイスをネットワークに接続するために必要なものです。
+
+あなたのプロジェクトにはすでに `nerves_init_gadget` が依存関係として存在しています。しかし、デフォルトではDHCPを使います（`config/config.exs` の中の `config :nerves_init_gadget` の後の実行内容を確認してください）。静的IPアドレスを持つ方が簡単です。
+
+静的ネットワークを設定するには、 `config/config.exs` に以下の行を追加する必要があります:
+
+```
+# Statically assign an address
+config :nerves_network, :default,
+  eth0: [
+    ipv4_address_method: :static,
+    ipv4_address: "192.168.88.2",
+    ipv4_subnet_mask: "255.255.255.0",
+    nameservers: ["8.8.8.8", "8.8.4.4"]
+  ]
+```
+
+この設定は有線接続用です。無線接続を使用したい場合は、[Nerves network documentation](https://github.com/nerves-project/nerves_network#wifi-networking) を参照してください。
+
+ここであなたのローカルネットワークパラメータを使う必要があることに注意してください。私のネットワークでは割り当てられていないIP `192.168.88.2`があるので、これを使用します。しかし、あなたの場合、IPアドレスは違うかもしれません。
+
+これを変更した後、私たちは `mix firmware.burn` を通してファームウェアの変更されたバージョンを書き込んで、それから新しいカードで装置を起動する必要があります。
+
+デバイスに電源を入れると、 `ping`を使ってオンラインになるのを見ることができます。
+
+```
+Request timeout for icmp_seq 206
+Request timeout for icmp_seq 207
+64 bytes from 192.168.88.2: icmp_seq=208 ttl=64 time=2.247 ms
+64 bytes from 192.168.88.2: icmp_seq=209 ttl=64 time=2.658 ms
+```
+
+この出力は、デバイスがネットワークから到達可能になったことを意味します。
+
+## ネットワークファームウェアの書き込み
+
+これまでのところ、私達はSDカードに書き込んで、物理的にそれらを私達のハードウェアにロードしてきました。これは最初は問題ありませんが、ネットワーク経由で更新する方が簡単です。`nerves_firmware_ssh` パッケージはまさにそれをしてくれます。デフォルトでは、すでにプロジェクトに存在し、ディレクトリ内のSSHキーを自動検出して見つけるように設定されています。
+
+ネットワークファームウェアアップデート機能を使うためには、 `mix firmware.gen.script` を通してアップロードスクリプトを生成する必要があります。このコマンドはファームウェアを更新するために実行できる新しい `upload.sh` スクリプトを生成します。
+
+前の手順を実行してネットワークが機能している場合は、問題ないので次に進みます。
+
+設定を更新するための最も簡単な方法は `mix firmware && ./upload.sh 192.168.88.2` を使うことです。最初のコマンドは更新されたファームウェアを作成し、二番目のコマンドはそれをネットワークにプッシュしてデバイスを再起動します。 SDカードをデバイスに出し入れする必要がなくなります。
+
+
+_ヒント： `ssh 192.168.88.2` はアプリのコンテキストでデバイスにIExシェルを提供します。_
+
+_トラブルシューティング：ホームフォルダにsshキーが存在しない場合は、 `No SSH public keys found in ~/.ssh.` というエラーが表示されます。この場合、 `ssh-keygen` を実行してネットワークアップデート機能を使用するためにファームウェアを再書き込みする必要があります。_
+
+## LEDのコントロールを設定する
+
+LEDとやり取りするには、[nerves_leds](https://github.com/nerves-project/nerves_leds) パッケージをインストールする必要があります。これは、`mix.exs` ファイルに `{:nerves_leds, "~> 0.8", targets: @all_targets},` を追加することによって行われます。 
+
+依存関係を設定したら、特定のデバイスのLEDリストを設定する必要があります。例えば、すべてのRaspberry Piモデルのために、ただ一つのLEDである `led0` が搭載されています。 `config/config.exs` に `config :nerves_leds, names: [green: "led0"]` を追加して使用しましょう。
+
+他のデバイスについては、[corresponding part of the nerves_examples project](https://github.com/nerves-project/nerves_examples/tree/master/hello_leds/config) を参照してください。
```

Comment:
```suggestion
他のデバイスについては、[nerves_examples プロジェクトで対応する部分](https://github.com/nerves-project/nerves_examples/tree/master/hello_leds/config) を参照してください。
```

It's just english. same reason with https://github.com/elixirschool/elixirschool/pull/1889#discussion_r294030195